src/orderbook.js

```diff
@@ -1197,6 +1336,44 @@ OrderBook.prototype.computeAutobridgedOffers = function() {
   this._offersAutobridged = autobridgeCalculator.calculate();
 };
 
+OrderBook.prototype.computeAutobridgedOffersWrapper = function() {
+  var startTime = Date.now();
+  this.computeAutobridgedOffers();
+  this.mergeDirectAndAutobridgedBooks();
+  var lasted = (Date.now() - startTime);
+  
+  const newMult =
+    ((lasted * 2 / OrderBook.AUTOBRIDGE_CALCULATE_THROTTLE_TIME) << 0) + 1;
+  if (newMult !== this._autobridgeThrottleTimeMultiplier) {
+    this._autobridgeThrottleTimeMultiplier = newMult;
+    this.createDebouncedOffersWrapper();
+  }
+}
+
+OrderBook.prototype.createDebouncedOffersWrapper = function() {
+  const m = this._autobridgeThrottleTimeMultiplier;
+  this.computeAutobridgedOffersThrottled = 
+    _.debounce(
```

Comment:
> You're saying that we just need to re-request orderbook from server after each ledger_closed event?

Not necessarily, you can just request the transactions in that ledger and do the processing in the same way. But even that's not necessary either, you can just buffer the stream until you get the ledger closed event. Actually it looks like @sublimator was working on this already:

```
+    // We'll wait until the ledger close message which always comes AFTER all
+    // the {"type: "transaction", ... } messages in a well specified (if not
+    // documented) order.
```
https://github.com/ripple/ripple-lib/pull/529/files

I'm not sure why this pull request was closed.